src/rviz_visual_tools.cpp

```diff
@@ -1566,6 +1566,32 @@ bool RvizVisualTools::publishPath(const EigenSTL::vector_Vector3d &path, const c
   return triggerInternalBatchPublishAndDisable();
 }
 
+bool RvizVisualTools::publishPath(const std::vector<Eigen::Vector3d> &path, const std::vector<colors> &colors, const double radius,
+                                  const std::string &ns)
+{
+  if (path.size() < 2)
+  {
+    ROS_WARN_STREAM_NAMED(name_, "Skipping path because " << path.size() << " points passed in.");
+    return true;
+  }
+
+  if (path.size() != colors.size())
+  {
+    ROS_ERROR_STREAM_NAMED(name_, "Skipping path because " << path.size() << " different from " << colors.size() << ".");
+    return false;
+  }
+
+  // Batch publish, unless it is already enabled by user
+  enableInternalBatchPublishing(true);
+
+  // Create the cylinders
+  for (std::size_t i = 1; i < path.size(); ++i)
+    publishCylinder(path[i - 1], path[i], colors[i], radius, ns);
```

Comment:
An alternative that might be more efficient is to use the [LINE_STRIP or LINE_LIST](http://wiki.ros.org/rviz/DisplayTypes/Marker#Line_Strip_.28LINE_STRIP.3D4.29) marker type, which I just saw can support this:

> In visualization 1.1+ will also optionally use the colors member for per-vertex color.

The downside is that those lines have a bug in Rviz s.t. when you rotate around the viewfinder in Rviz, they become invisible. I believe this is some kind of ogre settings problem that needs to be fixed, but I tried once and got overwhelmed by my lack of ogre knowledge. For this reason, I often use cylinders. Thoughts?